app/assets/javascripts/oxalis/model/reducers/flycam_reducer.js

```diff
@@ -171,13 +173,13 @@ function FlycamReducer(state: OxalisState, action: Action): OxalisState {
     case "SET_POSITION": {
       // cannot use M4x4.clone because of immutable-seamless
       const matrix = cloneMatrix(state.flycam.currentMatrix);
-      if (action.dimensionToSkip !== 0) {
+      if (action.dimensionToSkip !== 0 && !isNaN(matrix[12])) {
```

Comment:
I think you're checking the wrong value here. `matrix` is from the store and action.position should be checked.
```suggestion
      const { position } = action;
      if (action.dimensionToSkip !== 0 && !isNaN(position[0])) {
```

Reply:
Same for the next two checks.